tests/pass/move-data-across-await-point.rs

```diff
@@ -0,0 +1,78 @@
+use std::future::Future;
+
+// This test:
+// - Compares addresses of non-Copy data before and after moving it
+// - Writes to the pointer after it has moved across the await point
+async fn data_moved_async() {
+    // Vec<T> is not Copy
+    let mut x: Vec<u8> = vec![2];
+    let raw_pointer = &mut x as *mut Vec<u8>;
+    helper_async(x, raw_pointer).await;
+    unsafe {
+        assert_eq!(*raw_pointer, vec![3]);
+        // Drop to prevent leak.
+        std::ptr::drop_in_place(raw_pointer);
+    }
+}
+
+async fn helper_async(mut data: Vec<u8>, raw_pointer: *mut Vec<u8>) {
```

Comment:
Please move this function to inside `data_moved_async`, they seem tightly coupled.